hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -491,61 +492,47 @@ func (r *HostedClusterReconciler) listNodePools(clusterNamespace, clusterName st
 	return filtered, nil
 }
 
-func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request) (*ctrl.Result, error) {
-	targetNamespace := req.Name
-
-	var result *ctrl.Result
-	result = nil
-	nodePools, err := r.listNodePools(req.Namespace, req.Name)
+// delete will try to delete the HostedCluster and all its associated resources.
+// When the deletion is complete, delete will return true, indicating the HostedCluster
+// can safely be finalized. Otherwise returns false.
+func (r *HostedClusterReconciler) delete(ctx context.Context, hc *hyperv1.HostedCluster) (bool, error) {
+	nodePools, err := r.listNodePools(hc.Namespace, hc.Name)
 	if err != nil {
-		return result, fmt.Errorf("failed to get nodePools by cluster name for cluster %q: %w", req.Name, err)
+		return false, fmt.Errorf("failed to get nodePools by cluster name for cluster %q: %w", hc.Name, err)
 	}
 
 	for key := range nodePools {
 		if err := r.Delete(ctx, &nodePools[key]); err != nil && !apierrors.IsNotFound(err) {
-			return result, fmt.Errorf("failed to delete nodePool %q for cluster %q: %w", nodePools[key].GetName(), req.Name, err)
+			return false, fmt.Errorf("failed to delete nodePool %q for cluster %q: %w", nodePools[key].GetName(), hc.Name, err)
 		}
 	}
 
-	r.Log.Info("Deleting Cluster", "clusterName", req.Name, "clusterNamespace", targetNamespace)
+	r.Log.Info("Deleting Cluster", "clusterName", hc.Name, "clusterNamespace", hc.Name)
 	cluster := &capiv1.Cluster{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      req.Name,
-			Namespace: targetNamespace,
+			Name:      hc.Name,
```

Comment:
subjective nit: having a variable `targetNamespace` storing `hc.Name` and use it in these references makes it more readable an intuitive to me. Also easier to not get mistaken if more need to be added.

Reply:
The manifests.HostedControlPlaneNamespace struct encapsulates the namespace resource's details; we shouldn't be leaking out the naming strategy anywhere, and could instead do something like:

```
targetNamespace := manifests.HostedControlPlaneNamespace{HostedCluster: hcluster}.Build()
capiCluster := controlplaneoperator.CAPICluster{...}.Build()
```

Reply:
Let's fix up the hard-coded naming assumptions separately